llvm/lib/Demangle/MicrosoftDemangle.cpp

```diff
@@ -2266,15 +2296,17 @@ Demangler::demangleTemplateParameterList(std::string_view &MangledName) {
     } else if (consumeFront(MangledName, "$$C")) {
       // Type has qualifiers.
       TP.N = demangleType(MangledName, QualifierMangleMode::Mangle);
-    } else if (llvm::itanium_demangle::starts_with(MangledName, "$1") ||
-               llvm::itanium_demangle::starts_with(MangledName, "$H") ||
-               llvm::itanium_demangle::starts_with(MangledName, "$I") ||
-               llvm::itanium_demangle::starts_with(MangledName, "$J")) {
+    } else if (startsWithOrWithoutPrefix(MangledName, "$1", IsAutoNTTP, 1) ||
```

Comment:
I am not super happy with this fix but it was the cleanest way I could come up with to support `$M` within the current infrastructure with minimal code changes.

Once I fix the rest of the name mangling issues in clang I need to come back here and most likely refactor how we demangle the template parameter list with a larger holistic view of the different mangling schemes MSVC has.